src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -183,10 +190,60 @@ public static ModuleTag parseModuleTag(String tag) throws ParseException {
         requireNonNull(tag);
         String trimmedTag = tag.trim();
         ArrayList<String> args = parseMoreModules(trimmedTag);
-        if (!ModuleTag.isValidTagName(args.get(0))) {
+
+        if (args.size() == 1) {
+            return parseModuleTagFromSingle(args.get(0));
+        }
+
+        if (args.size() != 4) {
+            throw new ParseException(ModuleTag.MESSAGE_CONSTRAINTS);
+        }
+
+        String moduleCode = args.get(0);
+        Day day = parseDay(args.get(1));
+        LocalTime startTime = parseLocalTime(args.get(2));
+        LocalTime endTime = parseLocalTime(args.get(3));
+        TimeBlock timeBlock = new TimeBlock(startTime, endTime, day);
+
+        Lesson lesson = new Lesson(moduleCode, Location.NUS, timeBlock);
+
+        return new ModuleTag(moduleCode, lesson);
+    }
+
+    private static ModuleTag parseModuleTagFromSingle(String tag) throws ParseException {
+        if (!ModuleTag.isValidTagName(tag)) {
             throw new ParseException(ModuleTag.MESSAGE_CONSTRAINTS);
         }
-        return new ModuleTag(args);
+        return new ModuleTag(tag);
+    }
+
+    /**
+     * Tries to partial match a day string.
+     * MON will output MONDAY.
+     */
+    public static Day parseDay(String dayAsStr) throws ParseException {
+        String upperDayAsStr = dayAsStr.toUpperCase();
+        for (Day day : Day.values()) {
+            if (day.toString().contains(upperDayAsStr)) {
+                return day;
+            }
+        }
+        throw new ParseException("Day is invalid");
+    }
+
+    /**
+     * Parses a local time and checks whether it is valid.
+     */
+    public static LocalTime parseLocalTime(String localTimeAsStr) throws ParseException {
+        try {
+            int hour = Integer.parseInt(localTimeAsStr);
+            if (!Arrays.asList(Timetable.START_TIMINGS).contains(hour)) {
```

Comment:
@russelltankaimin Could there be a method in `Timetable` that handles this?

Reply:
Seems like a TimeUtils thing. It does not make sense to have a `timetable.parseLocalTime(String)`.

Reply:
In that case, start timings needs to be stored in TimeUtils, instead of Timetable.